libraries/provider_dnsimple_record.rb

```diff
@@ -29,12 +29,17 @@ def load_current_resource
         @current_resource.domain(@new_resource.domain)
         @current_resource.type(@new_resource.type)
 
-        records = dnsimple_client.zones.all_zone_records(dnsimple_client_account_id,
-                                                         @new_resource.domain)
-        @existing_record = records.data.detect do |record|
-          (record.name == @new_resource.record_name) && (record.type == @new_resource.type)
+        record_filter = { name: @new_resource.record_name, type: @new_resource.type }
+        records = dnsimple_client.zones.zone_records(dnsimple_client_account_id,
+                                                     @new_resource.domain,
+                                                     filter: record_filter)
+
+        if records.data.length > 1
+          Chef::Log.info "DNSimple: Multiple records found #{@new_resource.record_name}.#{@new_resource.domain}" \
+            " with type #{@new_resource.type}. This cookbook only modifies the first record."
         end
 
+        @existing_record = records.data.sort.first
```

Comment:
```suggestion
        @existing_record = records.data.first
```

I'm not sure why `sort` is called here. We should rely on API sorting (it will return the first by ID). Sorting on the Ruby data will have unexpected behavior (I'm not even sure what it is sorting by... Ruby object ID?).

Reply:
Sounds good! I was unsure we had a gaurenteed order as part of the API contract!